src/main/java/modtrekt/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,95 @@
+import modtrekt.logic.commands.exceptions.CommandException;
+import modtrekt.logic.parser.CliSyntax;
+import modtrekt.model.Model;
+import modtrekt.model.module.ModCode;
+import modtrekt.model.task.Deadline;
+import modtrekt.model.task.Description;
+import modtrekt.model.task.Task;
+
+
+/**
+ * Edits a task in the task book.
+ */
+public class EditTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+    public static final String COMMAND_IDENTIFIER = "-t";
+
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the task. \n"
+            + "Usage: " + COMMAND_WORD + " " + COMMAND_IDENTIFIER + " <INDEX>"
+            + " [" + CliSyntax.PREFIX_MOD_CODE + "<MODULE_CODE> " + "] "
+            + " [" + CliSyntax.PREFIX_DEADLINE + "<DEADLINE> " + "] "
+            + " [" + CliSyntax.PREFIX_TASK_DESCRIPTION + "<TASK DESCRIPTION> " + "] ";
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Task successfully edited: %1$s";
+
+    private final Index targetIndex;
+
+    private final ModCode targetModule;
+
+    private final LocalDate targetDeadline;
+
+    private final Description targetDesciption;
+
+    /**
+     * Creates an EditTaskCommand to edit the specified {@code Task}
+     */
+    public EditTaskCommand(Index targetIndex, ModCode targetModule, LocalDate targetDeadline,
+                           Description targetDesciption) {
+        this.targetIndex = targetIndex;
+        this.targetModule = targetModule;
+        this.targetDeadline = targetDeadline;
+        this.targetDesciption = targetDesciption;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToEdit = lastShownList.get(targetIndex.getZeroBased());
+
+        boolean archived = taskToEdit.isArchived();
+        ModCode code = targetModule != null ? targetModule : taskToEdit.getModule();
+        if (!model.hasModuleWithModCode(code)) {
+            throw new CommandException(String.format("Module code %s does not exist.",
+                    code.toString()));
+        }
+        Description description = targetDesciption != null ? targetDesciption : taskToEdit.getDescription();
+        LocalDate deadline = targetDeadline != null
+                ? targetDeadline
+                : taskToEdit instanceof Deadline
+                ? ((Deadline) taskToEdit).getDueDate()
+                : null;
+
+        Task newTask = deadline != null
+                ? new Deadline(description, code, deadline, archived)
+                : new Task(description, code, archived);
+
+        model.deleteTask(taskToEdit);
+        model.addTask(newTask);
+        model.updateModuleTask(newTask);
```

Comment:
this line is redundant, addTask calls this method already

Reply:
removed